src/modules/aws_vpc/mappers/route_table.ts

```diff
@@ -0,0 +1,266 @@
+import { EC2, paginateDescribeRouteTables } from '@aws-sdk/client-ec2';
+import { CreateRouteTableCommandInput } from '@aws-sdk/client-ec2/dist-types/commands/CreateRouteTableCommand';
+import {
+  Route as AwsRoute,
+  RouteTable as AwsRouteTable,
+} from '@aws-sdk/client-ec2/dist-types/models/models_1';
+
+import { AWS, crudBuilderFormat, paginateBuilder } from '../../../services/aws_macros';
+import { getCloudId } from '../../../services/cloud-id';
+import { findDiff } from '../../../services/diff';
+import { Context, Crud2, MapperBase } from '../../interfaces';
+import { Route, RouteTable } from '../entity';
+import { AwsVpcModule } from '../index';
+import { convertTagsForAws, convertTagsFromAws, eqTags } from './tags';
+
+export class RouteTableMapper extends MapperBase<RouteTable> {
+  module: AwsVpcModule;
+  entity = RouteTable;
+  equals = (a: RouteTable, b: RouteTable) => {
+    return (
+      a.vpc.vpcId === b.vpc.vpcId &&
+      this.eqListItems<Route>(this.eqRoute, a.routes, b.routes) &&
+      eqTags(a.tags, b.tags)
+    );
+  };
+
+  eqListItems<T>(eq: (a: T, b: T) => boolean, a?: T[], b?: T[]) {
+    const subset1 = !!a?.every(a1 => !!b?.find(a2 => eq(a1, a2)));
+    const subset2 = !!b?.every(a1 => !!a?.find(a2 => eq(a1, a2)));
+    return subset1 && subset2;
+  }
```

Comment:
This is actually subtly broken, believe it or not. The following code should show it:

```ts
function eqListItems<T>(eq: (a: T, b: T) => boolean, a?: T[], b?: T[]) {
  const subset1 = !!a?.every(a1 => !!b?.find(a2 => eq(a1, a2)));
  const subset2 = !!b?.every(a1 => !!a?.find(a2 => eq(a1, a2)));
  return subset1 && subset2;
}

const firstThreeWholeNumbers = [1, 2, 3];
const firstFourFibonacciNumbers = [1, 1, 2, 3];

console.log(eqListItems((a, b) => a === b, firstThreeWholeNumbers, firstFourFibonacciNumbers));
```

Which in `ts-node` gives me:

```sh
damocles@zelbinion:~/alantech/iasql-engine(metadata-rpcs)$ ./node_modules/.bin/ts-node
> function eqListItems<T>(eq: (a: T, b: T) => boolean, a?: T[], b?: T[]) {
... const subset1 = !!a?.every(a1 => !!b?.find(a2 => eq(a1, a2)));
... const subset2 = !!b?.every(a1 => !!a?.find(a2 => eq(a1, a2)));
... return subset1 && subset2;
... }
undefined
> const firstThreeWholeNumbers = [1, 2, 3];
undefined
> const firstFourFibonacciNumbers = [1, 1, 2, 3];
undefined
> console.log(eqListItems((a, b) => a === b, firstThreeWholeNumbers, firstFourFibonacciNumbers));
true
undefined
>
```

When those are clearly two different lists.

It *probably* doesn't matter, but first early returning if the list lengths are different, then using a sorting function to deterministically sort the two arrays, and then scan the arrays one-at-a-time using the same sorting function to confirm that they both sort into the same spot (the JS-style sorting function returns zero) would be more correct *and* have a smaller Big-O notation than this implementation (the current one being `O(n^2)` while the described implementation would be `O(n*log(n))` (though for small `n` it would definitely be slower).

Just adding a `length` comparison check on the current implementation wouldn't work, because it would still pass for `[1, 1, 2, 3]` vs `[1, 2, 2, 3]`, for instance.

But if you don't allow duplicates in your comparison, then you can probably get away with what's currently there, perhaps with a pre-check scan on both arrays that none of the arrays have any elements that match with any other elements in the array (harder to do functionally, though, but iteratively on this one with separate `i` and `j` counter variables could speed it up by letting you do the four needed comparisons at the same time, instead of in separate loops, and allow early exiting on any of the four conditions failing).

Reply:
I was proven wrong mathematically :))